src/makie.jl

```diff
@@ -209,18 +225,293 @@ function new_cycle_theme()
+    fig
+end
+
+function mixed_mode_layout()
+    CairoMakie.activate!() # hide
+    Random.seed!(123)
+    longlabels = ["$(today() - Day(1))","$(today())","$(today() + Day(1))"]
+
+    fig = Figure(resolution = (600,400), fontsize = 12,
+        backgroundcolor = :grey90, font="CMU Serif")
+    ax1 = Axis(fig[1,1])
+    ax2 = Axis(fig[1,2],xticklabelrotation =pi/2,alignmode =Mixed(bottom=0),
+        xticks = ([1,5,10], longlabels))
+    ax3 =  Axis(fig[2,1:2])
+    ax4 =  Axis(fig[3,1:2])
+
+    lines!(ax1, 1:10, rand(10))
+    lines!(ax2, 1:10, rand(10))
+    lines!(ax3, 1:10, rand(10))
+    lines!(ax4, 1:10, rand(10))
+    hidexdecorations!(ax3, ticks = false, grid = false)
+
+    Box(fig[2:3,1:2, Right()], color = (:slateblue1, 0.35))
+    Label(fig[2:3,1:2, Right()],"protrusion",rotation = pi/2,textsize = 14,
+        padding = (3,3,3,3))
+    Label(fig[1,1:2, Top()], "Mixed alignmode", textsize = 16,
+        padding = (0,0,15,0))
+    
+    colsize!(fig.layout, 1, Auto(2))
+    rowsize!(fig.layout, 2, Auto(0.5))
+    rowsize!(fig.layout, 3, Auto(0.5))
+    rowgap!(fig.layout, 1, 15)
+    rowgap!(fig.layout, 2, 0)
+    colgap!(fig.layout, 5)
+    fig
+end
+
+function nested_sub_plot!(fig)
+    color = rand(RGBf0)
+    ax1 = Axis(fig[1,1], backgroundcolor = (color,0.25))
+    ax2 = Axis(fig[1,2], backgroundcolor = (color,0.25))
+    ax3 = Axis(fig[2,1:2], backgroundcolor = (color,0.25))
+    ax4 = Axis(fig[1:2,3], backgroundcolor = (color,0.25))
+    return (ax1,ax2,ax3,ax4)
+end
+
+function main_figure()
+    Random.seed!(123) # hide
+    CairoMakie.activate!() # hide
+    fig = Figure()
+    Axis(fig[1,1])
+    nested_sub_plot!(fig[1,2])
+    nested_sub_plot!(fig[1,3])
+    nested_sub_plot!(fig[2,1:3])
+    fig
+end
+
+function nested_Grid_Layouts()
+    CairoMakie.activate!() # hide
+    fig = Figure(backgroundcolor = RGBf0(0.96, 0.96, 0.96),)    
+    ga = fig[1, 1] = GridLayout()
+    gb = fig[1, 2] = GridLayout()
+    gc = fig[1, 3] = GridLayout()
+    gd = fig[2, 1:3] = GridLayout()
+    gA = Axis(ga[1,1])
+    nested_sub_plot!(gb)
+    axsc = nested_sub_plot!(gc)
+    nested_sub_plot!(gd)
+
+    [hidedecorations!(axsc[i],grid=false,ticks=false) for i in 1:length(axsc)]
+
+    colgap!(gc, 5)
+    rowgap!(gc, 5)
+    rowsize!(fig.layout, 2, Auto(0.5))
+    colsize!(fig.layout, 1, Auto(0.5))
+    fig
+end
+
+function add_box_inset(fig;left=100,right=250,bottom = 200,top= 300, 
+        bgcolor=:grey90)
+    # https://discourse.julialang.org/t/makie-inset-axes-and-their-drawing-order/60987 # hide
+    inset_box = Axis(fig, bbox = BBox(left, right, bottom, top),
+        xticklabelsize = 12, yticklabelsize=12, backgroundcolor = bgcolor)
+    
+    # bring content upfront
+    translate!(inset_box.scene, 0, 0, 10)
+    elements = keys(inset_box.elements)
+    for ele in elements
+        if ele == :xaxis || ele == :yaxis
+            nothing
+        else 
+            translate!(inset_box.elements[ele], 0, 0, 9)
+        end
+    end
```

Comment:
isn't this the same as something like
```
filtered = filter(e -> ele != :xaxis && ele !=:yaxis , elements)
foreach(translate!(inset_box.elements[ele], 0, 0, 9), filtered)
```
or with a for loop instead of higher order functions:
```
for ele in elements
    if ele != :xaxis && ele !=:yaxis
        translate!(inset_box.elements[ele], 0, 0, 9)
    end
end
```

Reply:
yes, more or less (I few typos in your suggestion 😄 ). They are the same, but I prefer the first suggestion since you already mentioned the filter option, I think, somewhere. Plus, is easy to read. I will update the code with that.